### PR TITLE
core/txpool/legacypool: move locals-handling out of pool

### PR DESCRIPTION
**Pull request by holiman**
This is work in progress.
This PR disables all locals-handling within the legacy transaction pool (it doesn't remove the code (yet) just sets local to false in the txpool during submission).

In place of the locals, this PR instead utilizes a TxTracker, a tracker which, from time to time (every minute), checks on the set of transactions that it tracks, whether the transaction seems to have been forgotten and needs to be resubmitted to the pool.

This design makes the handling of locals be not part of the internals of the pool, which simplifies a lot since it removes a lot of cornercases where handling of locals/nonlocals use different rules.

### CODE REVIEW

File: eth/catalyst/api_test.go [hunk 1]
```diff
@@ -107,7 +107,7 @@ func TestEth2AssembleBlock(t *testing.T) {
 	if err != nil {
 		t.Fatalf("error signing transaction, err=%v", err)
 	}
-	ethservice.TxPool().Add([]*txpool.Transaction{{Tx: tx}}, true, false)
+	ethservice.TxPool().Add([]*txpool.Transaction{{Tx: tx}}, false, false)
```
Comment (holiman) [inline eth/catalyst/api_test.go:110]:
```suggestion
	ethservice.TxPool().Add([]*txpool.Transaction{{Tx: tx}}, true, false)
```

File: miner/worker_test.go [hunk 2]
```diff
@@ -194,8 +194,8 @@ func TestGenerateAndImportBlock(t *testing.T) {
 	w.start()
 
 	for i := 0; i < 5; i++ {
-		b.txPool.Add([]*txpool.Transaction{{Tx: b.newRandomTx(true)}}, true, false)
-		b.txPool.Add([]*txpool.Transaction{{Tx: b.newRandomTx(false)}}, true, false)
+		b.txPool.Add([]*txpool.Transaction{{Tx: b.newRandomTx(true)}}, false, false)
+		b.txPool.Add([]*txpool.Transaction{{Tx: b.newRandomTx(false)}}, fa;se, false)
```
Comment (holiman) [inline miner/worker_test.go:198]:
```suggestion
		b.txPool.Add([]*txpool.Transaction{{Tx: b.newRandomTx(true)}}, true, false)
		b.txPool.Add([]*txpool.Transaction{{Tx: b.newRandomTx(false)}}, true, false)
```

File: eth/api_backend.go [hunk 3]
```diff
@@ -294,7 +294,10 @@ func (b *EthAPIBackend) SubscribeLogsEvent(ch chan<- []*types.Log) event.Subscri
 }
 
 func (b *EthAPIBackend) SendTx(ctx context.Context, signedTx *types.Transaction) error {
-	return b.eth.txPool.Add([]*txpool.Transaction{&txpool.Transaction{Tx: signedTx}}, true, false)[0]
+	if locals := b.eth.localTxTracker; locals != nil {
+		locals.Track(signedTx)
+	}
+	return b.eth.txPool.Add([]*txpool.Transaction{&txpool.Transaction{Tx: signedTx}}, false, false)[0]
```
Comment (holiman) [inline eth/api_backend.go:300]:
```suggestion
	return b.eth.txPool.Add([]*txpool.Transaction{&txpool.Transaction{Tx: signedTx}}, true, false)[0]
```

**Comment by holiman** [comment]
The follow-up to this would be something like this:https://github.com/ethereum/go-ethereum/commit/d93d6ac6210fae3bf782a5a3aff5d5a0606f13f3 -- removing the handling of locals from the pools. (+ need to fix up the tests also)

**Comment by rjl493456442** [comment]
IIUC, the tracker will only tracks the local transactions, but try to resubmit them time to time instead of having special rules in txpool, right?

**Comment by holiman** [comment]
> IIUC, the tracker will only tracks the local transactions, but try to resubmit them time to time instead of having special rules in txpool, right?

Yes, sounds correct.

**Comment by holiman** [comment]
Triage discussion: the locals-pool should be implemented as a full subpool instead of this tracker-approach.

**Pull request closed without merging**